packages/ui-react/src/components/UpgradeSubscription/UpgradeSubscription.tsx

```diff
@@ -23,6 +25,19 @@ const UpgradeSubscription: React.FC<Props> = ({ type, onCloseButtonClick }: Prop
   const title = t(`checkout.upgrade_${type}`);
   const message = t(`checkout.upgrade_${type}_message`);
 
+  const typeToAnnounce: Record<Props['type'], AriaAnnouncerVariant> = useMemo(
+    () => ({
+      error: 'error',
+      success: 'success',
+      pending: 'info',
+    }),
+    [],
+  );
+
+  useEffect(() => {
+    announcer(message, typeToAnnounce[type]);
+  }, [announcer, message, type, typeToAnnounce]);
```

Comment:
Minor, but declaring the const inside the useEffect looks a bit cleaner, and prevents 1 render:
```javascript
useEffect(() => {
  const typeToAnnounce = { error: 'error', success: 'succes', pending: 'info'}[type];

  announcer(message, typeToAnnounce);
}, [announcer, message, type]);

```

Reply:
Good point, ill move the declaration of the const inside the useEffect.